test/cmocka/src/notifier_mocks.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/notifier.h>
+#include <sof/list.h>
+#include <sof/common.h>
+
+struct callback_handle {
+	void *receiver;
+	void *caller;
+	void (*cb)(void *arg, enum notify_id, void *data);
+	struct list_item list;
+};
+
+static struct notify *_notify;
+
+struct notify **arch_notify_get(void)
+{
+	int i;
+
+	if (!_notify) {
+		_notify = rzalloc(0, 0, sizeof(*_notify));
```

Comment:
Is this mock function used like that in other unit tests? Feels slightly weird to me to not just use malloc here.

Reply:
Requesting change on this one, looked more into it: The other tests and mocks actually indeed call `malloc` directly.